include/libdnf5/base/goal_elements.hpp

```diff
@@ -169,6 +169,9 @@ struct GoalJobSettings : public ResolveSpecSettings {
     /// If set to true, group operations (install / remove) will only work
     /// with the group itself, but will not add to the transaction any packages.
     bool group_no_packages{false};
+    /// If set to true, environment operations (install / remove) will only work
+    /// with the environment itself, but will not add to the transaction any groups.
+    bool environment_no_groups{false};
```

Comment:
Please don't take it as no, but I think that this will brake ABI. I think we discussed it recently and my recommendation would be to open a discuss again and discuss all option that we have.